### PR TITLE
Add Jenkins pipeline configuration for develop branch

### LINKED ISSUE #445: [FEAT] Add continuous integration for the develop branch
### Feature description

The idea of adding Continuous Integration for the master branch is for testing all the examples against the latest weekly or unreleased version of RTI Connext DDS. The purpose of the pipeline will be to build the examples and perform the static analysis among them, and in the future, we will add a Valgrind stage to find memory leaks.

### PR DESCRIPTION
**Pull request by ManuelJNunez**
### Summary

I added a new `Dockerfile` and a `Jenkinsfile` to configure the Continuous Integration for develop branch. I also added a new `Python` script and `Markdown` templates to generate the check details text.

### Details and comments

close #445 

### Checks

-   [x] I have updated the documentation accordingly.
-   [x] I have read the [CONTRIBUTING](https://github.com/rticommunity/rticonnextdds-examples/blob/develop/CONTRIBUTING.md) document.

### CODE REVIEW

**Comment by lulivi** [comment]
@ManuelJNunez please update the README.md in the resources/ci_cd direcotry

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,150 @@
+def DETAILS_URL="https://www.rti.com"
```
Comment (lulivi) [inline Jenkinsfile:1]: @jpovedano @fgaranda what do you think about this URL? Maybe we should put the community portal URL or something like that?

This appears when you click on the "details" link when the action is launched:

image

Comment (ManuelJNunez) [reply Jenkinsfile:1]: My idea was to define the GitHub App's URL as the details URL once it is created. So maybe I will update it in the future.

Comment (ManuelJNunez) [reply Jenkinsfile]: The DETAILS_URL was changed to https://www.community.rti.com/

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,150 @@
+def DETAILS_URL="https://community.rti.com/"
```
Comment (lulivi) [inline Jenkinsfile:1]: Please add the license header here and that's all

**Review by lulivi — CHANGES_REQUESTED** [review]

**Review by lulivi — DISMISSED** [review]

**Review by lulivi — DISMISSED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# (c) 2021 Copyright, Real-Time Innovations, Inc.  All rights reserved.
```
Comment (fgaranda) [inline Dockerfile:1]: Use permissive copyright heading, see below in the Jenkinsfile

**Review by fgaranda — CHANGES_REQUESTED** [review]

**Review by lulivi — APPROVED** [review]

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+                                    "props": "rti.artifact.architecture=x64Linux4gcc7.3.0;rti.artifact.kind=staging",
+                                    "sortBy": ["created"],
+                                    "sortOrder": "desc",
+                                    "limit": 1,
+                                    "flat": true
+                                }]
+                            }''',
+                        )
+
+                        sh 'tar zxvf connextdds-staging-x64Linux4gcc7.3.0.tgz unlicensed/'
```
Comment (jpovedano) [inline Jenkinsfile:74]: Add a reference to https://www.jfrog.com/jira/browse/HAP-1154, that is the reason why we cannot use the `explode` functionality

**Review by jpovedano — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (ManuelJNunez) [reply Jenkinsfile]: The license header was added. You are reviewing an old commit.

File: resources/ci_cd/jenkins_output.py [hunk 5]
```diff
@@ -46,22 +46,33 @@ def main():
...
-            subprocess.run(["clang", "-dumpversion"], capture_output=True)
+            subprocess.run(
+                ["clang", "-dumpversion"], capture_output=True, check=True
+            )
             .stdout[:-1]
             .decode("utf-8")
+            or "-"
         )
+
         extra_rows = f"| CLANG Version | {clang_version} |\n"
```
Comment (lulivi) [inline resources/ci_cd/jenkins_output.py:68]: Are you overwriting the variable extra_rows here? Maybe you meant:

```suggestion
        extra_rows = f"{extra_rows}| CLANG Version | {clang_version} |\n"
```

**Review by lulivi — CHANGES_REQUESTED** [review]

File: resources/ci_cd/jenkins_output.py (hunk 5 above)
Comment (ManuelJNunez) [reply resources/ci_cd/jenkins_output.py:68]: Good catch!

File: Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+                            publishChecks conclusion: 'CANCELED', detailsURL: DETAILS_URL,
+                                name: STAGE_NAME, title: 'Aborted', text: readFile("jenkins_output.md"),
+                                summary: ':no_entry: The static analysis was aborted'
+                        }
+                    }
+                }
+            }
+
+            post {
+                always {
```
Comment (jpovedano) [inline Jenkinsfile:172]: use `cleanup` instead of `always`

**Review by jpovedano — APPROVED** [review]

**Review by fgaranda — APPROVED** [review]

**Review by lulivi — APPROVED** [review]

**Pull request merged**